### PR TITLE
[MRG+1] FIX+ENH: Fixes for 3D plotting, src.plot()

### PR DESCRIPTION
**Pull request by Eric89GXL**
I was digging around in our 3D plotting and found a number of somewhat interrelated things to fix/update. Can break up if necessary.

This PR does a few things:

1. ~~Adds MCE example. (@agramfort can you make sure it actually makes sense?)~~
2. Refactors some 3D functions to use a shared private `_create_mesh_surf` function, and create meshes a consistent way.
3. Sets backface culling consistently.
4. Modifies `plot_trans` to allow `src` argument.
5. Adds `src.plot()`, which wraps to `plot_trans`, to plot source spaces. For surface source spaces:
    surf
    And for mixed/volume/discrete source spaces (here mixed):
    mixed

### CODE REVIEW

File: examples/inverse/plot_mce_inverse.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+###############################################################################
+# Run solver
+
+alpha = 50  # regularization parameter between 0 and 100 (100 is high)
+loose, depth = 0.2, 0.9  # loose orientation & depth weighting
+
+# Compute MCE: equivalent to unweighted MxNE on a single time point!
+# To solve MCE for multiple time points, you could loop through them
+# one-by-one since they are treated independently.
```
Comment (joewalter) [inline examples/inverse/plot_mce_inverse.py]: We should point out that computing MCE for multiple time points by running MxNE for each sample seperately applies a different regularization parameter per time point. Even though alpha is the same, alpha_max varies between samples and the resulting regularization parameter lambda = alpha / 100 * alpha_max  is different. The alternative is to have an MCE function in the sparse module, which takes a time series, computes alpha_max by taking the max over samples, and then either loops over samples with MxNE to compute the MCE estimate per sample or applies MCE solver, which is, however, not yet part of MNE (I think the first approach should be good enough)

Comment (agramfort) [inline examples/inverse/plot_mce_inverse.py]: also this is not completely MCE as MCE was using the orientation obtained from MNE to be able to use off the shelf optimization tools.

@Eric89GXL why do you think it is a good idea to have an example with "MCE like" solution?

**Comment by Eric89GXL** [comment]
People have asked me about using MCE, and wanted to know if they could do
it in MNE

**Comment by agramfort** [comment]
tell them to use MxNE :)

but maybe yes it's good for google indexing... I am just a bit concerned
that it's not really MCE unless you use fixed ori with this code.

maybe a bit of documentation editing is enough...

**Comment by joewalter** [comment]
Using our MxNE code on single time samples is close to the SSI method described here:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2612127/pdf/nihms-59224.pdf

Using a loose orientation constraint in combination with MCE-like methods was discussed in detail here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4086627/pdf/nihms-588456.pdf

So if we add this functionality we can add these references, eventhough we are using other solvers

**Comment by Eric89GXL** [comment]
> also this is not completely MCE as MCE was using the orientation obtained from MNE to be able to use off the shelf optimization tools.

I haven't ever worked with MCE so I'm a bit out of my element here. *Someone* told me that MCE was like applying MxNE on a per-time-point basis, so that's what I went for here. Clearly it's not that simple :)

> I am just a bit concerned that it's not really MCE unless you use fixed ori with this code.

So for a first pass could we only allow fixed-orientation forward solutions, and maybe at some point later add support for loose/free orientation according to the MNE estimate you describe above?

> Even though alpha is the same, alpha_max varies between samples and the resulting regularization parameter lambda = alpha / 100 * alpha_max is different. The alternative is to have an MCE function in the sparse module, which takes a time series, computes alpha_max by taking the max over samples

That sounds doable. It probably makes sense to have a separate function anyway.

So how about:

1. Make a new function in `inverse_sparse`, call it something like `minimum_current_estimate` (assuming we are actually implementing some variant of that algorithm)?

2. Add refs to relevant papers and describe what we are actually doing. (@joewalter I might need a PR into my branch for this sort of thing if you have time.)

3. Make the function use all time points for `alpha_max`.

4. Apply MxNE time point by time point using the given `alpha_max`.

Would that work?

**Comment by agramfort** [comment]
@Eric89GXL please remove the example on MCE for now. Keep it in a separate PR. Then +1 for merge

**Comment by Eric89GXL** [comment]
@agramfort done.

Someone else want to have a look?

**Comment by agramfort** [comment]
thinking about it I find the new plot of src space less clear than my yellow dots on the cortex... I least you could see the spacing. I am using this figure all the time for teaching forward and inverse modeling.

**Comment by Eric89GXL** [comment]
> thinking about it I find the new plot of src space less clear than my yellow dots on the cortex... I least you could see the spacing.

I know the one you mean. Unfortunately plotting it on the `inflated` surface (easiest way for teaching) isn't geometrically correct (the goal of these plots). It does give me some ideas about this a compromise, though, WDYT:

screenshot from 2017-02-13 12-59-42

screenshot from 2017-02-13 13-32-26

You don't get the normals anymore, but it was really hard to see them before anyway.

**Comment by agramfort** [comment]
thx @Eric89GXL

**Pull request merged**